.github/workflows/provider-compatibility.yml

```diff
@@ -10,12 +10,10 @@
 
 name: Provider compatibility across versions
 
-# NOTE: if this is being run on pull_request, it will **not** use the pull
```

Comment:
Please keep this comment.

This test fails way too often and debugging it is really difficult if this nugget is forgotten.

Reply:
But there is no point in running this job on pull request. I've removed the commented out pull_request as well. You have the workflow_dispatch: option to run it manually and there is the prov-compat-label.yml workflow to be run on pull_request which is fully equivalent.

Reply:
Agreed there is no point running this on PR.  However, it's the most obvious thing to try when debugging cross version issues.  Having a comment saying that it's pointless to run it on PR will avoid wasted time in the future.

Despite being the author of this test, this has stung me twice which is why I added the comment.  It will catch people in the future who are not so intimately acquainted with how it operates.

Please put the comment back.

Reply:
Better comment added.